crates/ruff/src/settings/rule_table.rs

```diff
@@ -26,6 +26,11 @@ impl RuleTable {
         self.enabled.contains_key(code)
     }
 
+    /// Returns whether any of the given rules should be checked.
+    pub fn any_enabled(&self, codes: &[&Rule]) -> bool {
+        codes.iter().any(|c| self.enabled.contains_key(c))
+    }
```

Comment:
Neat! 

We can improve the ergonomics (reduce the amount of `&` a caller must write) by: 

1. Change `Rule` to implement `Copy` (Rule is only 2 bytes large, which is cheaper than passing a reference that is 8 bytes) by changing the derive macro [here](https://github.com/charliermarsh/ruff/blob/aa51ecedc5ebbba5173350d45f0ea4ae9ae41df6/crates/ruff_macros/src/register_rules.rs#L32-L45)
2. Accept any iterable 
```suggestion
    pub fn any_enabled(&self, codes: impl IntoIterator<Item=Rule>) -> bool {
        codes.into_iter().any(|c| self.enabled.contains_key(c))
    }
```


You can then use it like this

```rust
let enforce_ambiguous_unicode_character = settings.rules.any_enabled([
        Rule::AmbiguousUnicodeCharacterString,
        Rule::AmbiguousUnicodeCharacterDocstring,
        Rule::AmbiguousUnicodeCharacterComment,
    ]);
```